JOSS/paper.md

```diff
@@ -0,0 +1,103 @@
+
+Choosing serialization formats such as CSV, JSON, BSON, and HDF5 are prudent choices for long term storage because they are application and language agnostic. Projects that evolve beyond an initial language or library decision can still load old experimental data.
+These formats often support a limited set of types, requiring applications to define and maintain serialization methods for their custom objects.
+
+Using formats such as JLD (Julia Data), Python pickles or MAT files work best for convenient saving of arbitrary objects. Unfortunately, these formats are often highly coupled to the software dependencies (e.g., language version, software packages), which makes restoring the data more challenging as time passes and the software evolves.
+
+JLSO.jl is a library for conveniently storing arbitrary Julia objects while maintaining restoration reliability. The JLSO (Julia Serialized Object) format stores metadata about the save environment alongside the arbitrarily serialized objects. This metadata includes the:
+
+1. Julia release version
+2. Object serialization and compression methods used
+3. Package environment states, expressed via the Julia Project.toml and Manifest.toml files
+
+Let's use a simple example to demonstrate why this metadata is valuable. The next few lines we'll load a JLSO files and try to load various objects.
+
+```julia
+julia> using Pkg, JLSO
+
+julia> jlso = read("../test/specimens/v3_julia_serialize_none.jlso", JLSOFile)
+JLSOFile([ZonedDateTime, DataFrame, Vector, DateTime, String, Matrix, Distribution]; version="3.0.0", julia="1.0.3", format=:julia_serialize, compression=:none, image="")
+
+# Examples of different julia objects that can be serialized
+julia> names(jlso)
+7-element Array{Symbol,1}:
+ :ZonedDateTime
+ :DataFrame
+ :Vector
+ :DateTime
+ :String
+ :Matrix
+ :Distribution
+
+julia> jlso[:Matrix]
+3×3 Array{Float64,2}:
+ 0.400348   0.892196   0.848164
+ 0.0183529  0.755449   0.397538
+ 0.870458   0.0441878  0.170899
+
+julia> jlso[:String]
+"Hello World!"
+
+julia> jlso[:DateTime]
+2018-01-28T00:00:00
+
+julia> jlso[:Distribution];
+[warn | JLSO]: KeyError: key Distributions [31c24e10-a181-5473-b8eb-7969acd0382f] not found
+
+julia> using Distributions
+ERROR: ArgumentError: Package Distributions not found in current path:
+- Run `import Pkg; Pkg.add("Distributions")` to install the Distributions package.
+
+Stacktrace:
+ [1] require(::Module, ::Symbol) at ./loading.jl:893
+```
+
+As you can see, we do not have Distributions installed and we also don't know which version of Distributions.jl we need to successfully deserialize that object. As of version 1.2.0, users can resolve these kinds of concerns by rebuilding the package environment used to save a JLSO file directly from the REPL.
```

Comment:
```suggestion
As you can see, we do not have Distributions installed and we also don't know which version of Distributions.jl we need to successfully deserialize that object.
Users can resolve these kinds of concerns by rebuilding the package environment used to save a JLSO file directly from the REPL.
```
We don't need to mention features not present in older versions, noone is going to use older versions.